typed-racket-lib/typed-racket/env/init-envs.rkt

```diff
@@ -324,6 +324,8 @@
        ,(and rest (type->sexp rest))
        (list ,@(map type->sexp kws))
        ,(type->sexp rng))]
+    [(Rest: tys )
+     `(make-Rest (list ,@(map type->sexp tys)))]
```

Comment:
no, we need the explicit list there since this is hackily generating syntax (otherwise the first type sexp would be in operator position instead of `list` in output)